main.go

```diff
@@ -155,7 +160,32 @@ func main() {
 			ServerName:   *serverName,
 		})
 	}
-	db, err := sql.Open("mysql", *dsn)
+
+	var finalDsn = *dsn
+	if *prompt {
+		dsnRegex := regexp.MustCompile(`(\w*):?\w*(@.+)`)
+		matches := dsnRegex.FindStringSubmatch(finalDsn)
+		if matches == nil {
+			fmt.Print("Incorrect format for dsn. Usage:\n")
+			flag.PrintDefaults()
+			os.Exit(1)
+		}
+
+		fmt.Print("Enter password: ")
+		// don't echo password to screen during input
+		bytePassword, err := terminal.ReadPassword(int(syscall.Stdin))
```

Comment:
Looking at [1] the `bytePassword` doesn't seem a suitable name. I think simply `password` would work fine.

[1] https://godoc.org/golang.org/x/crypto/ssh/terminal#Terminal.ReadPassword